src/dotnet/Chat.UI.Blazor/Services/ChatUIStateSync.cs

```diff
@@ -186,19 +187,21 @@ protected virtual async Task<Symbol> SyncRecordingStateImpl(CancellationToken ca
 
         if (recordingChatId == recorderChatId) {
             // The state is in sync
-            if (!recordingChatId.IsEmpty) {
-                if (await IsLanguageChanged().ConfigureAwait(false))
-                    await SyncRecorderState().ConfigureAwait(false); // We need to toggle the recording in this case
-            }
+            if (recordingChatId.IsEmpty)
+                return default;
+
+            if (await IsLanguageChanged().ConfigureAwait(false))
+                SyncRecorderState(); // We need to toggle the recording in this case
         } else if (recordingChatIdChanged) {
             // The recording was activated or deactivated
-            await SyncRecorderState().ConfigureAwait(false);
-            if (!recordingChatId.IsEmpty) {
-                // Update _lastLanguageId
-                await IsLanguageChanged().ConfigureAwait(false);
-                // Start recording = start realtime playback
-                await ChatUI.SetListeningState(recordingChatId, true).ConfigureAwait(false);
-            }
+            SyncRecorderState();
```

Comment:
Why did await disappear before SyncRecorderState?

Reply:
It returns `void` - actually it spawns a BackgroundTask